sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -100,7 +99,7 @@ case class AddJar(path: String) extends RunnableCommand {
     // returns the value of a thread local variable and its HiveConf may not be the HiveConf
     // associated with `executionHive.state` (for example, HiveContext is created in one thread
     // and then add jar is called from another thread).
-    hiveContext.executionHive.state.getConf.setClassLoader(newClassLoader)
+    hiveContext.executionHive.state.getConf.setClassLoader(hiveContext.libraryClassLoader)
```

Comment:
I think that this pr also suffers the same problem exposed in https://github.com/apache/spark/pull/9170. There was no test failure because we only set the mutable classloader when we actually do add jar (In #9170 we set the mutable classloader to `executionHive.state.getConf` whenever we need to use `withHiveState`).